### DISCUSSION TITLE
[Feature] path variable as search item

### DISCUSSION THREAD

**Discussion by Surfer2010** [immich-app/immich#6519] [Feature Request]
### The feature

I read a lot about people (including me) trying to import their old library as "external library" and struggling about their "event-file-folders not beeing transformed into album automatically". Not mentioning subfolders and so on.

this could be solved by adding the "file path" of every assset/picture to the searchable items. This way I could search for a known folder name and add the result to an album of my choice / create a new album.
This is far from automatic album creation, but a huge step bringing old file structures as external libraries to immich.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by BlipRanger** [comment]
In particular, it would be super helpful if the search featured the same glob style querying as the exclusion paths do.

**Comment by tztopi** [comment]
Also same issue here. As many folks the filename are generated by camera (IMG_xxxx, 20250101_xxxxxx.jpg, P202501.....) and only the parent directory carries the "album name". In many cases multi-depth directory name has the whole information. For example:
travel/2024/amsterdam_jan/IMG_.......jpg. Without proper full-path index search solution these photos hardly to find.
Another real example:
projects/work/20241205_server_rack/IMG_..... .jpg
I know many phones makes proper in-sync timestamp based filename (because it has NTP client sync), but most DSLRs are still using IMG_.... naming pattern.
Yes i know there are many external python script available to generate album name from directory. I dont want to create separate album records, all information exists on directory names.

Immich are great, the external path function also great, just please include the full original path to the search index for filename search function. Glob style search query also would be great as mentioned by @BlipRanger earlier.

**Comment by ffchung** [comment]
I want to have search folder name in search bar combine with other condition. Like People A & People B & Folder name with "badminton".

**Reply by kvalev** [reply]
This is now possible with this PR https://github.com/immich-app/immich/pull/16173

**Reply by luziusalbin** [reply]
any chance to get something out of this PR to make the path variable be able to search?

**Comment by baumaeschi** [comment]
Yes, I can only support this feature request. This has been my most wanted feature for about two years or so. But no one from the Immich team seems interested in this. Or am I wrong? Is this on a future release map?
Want to mention, that photoprism has implemented this folder path search. This is the only feature of photoprism that makes me not switch completely to Immich.
Addition: This should be a feature for external Libraries! Add the folder names as keys for searching pictures, videos, etc.

**Comment by dfldylan** [comment]
I totoly agree

**Comment by luziusalbin** [comment]
Adding myself here as a total fan of the request!

I work in an environment of 15-20 people and everyone is using our photos, which is at this moment still windows-explorer standard usage and our paths are looking oftne like this: "P:/Photos/Projects/City, Customer-Name/2023/Measurment/..."

Some people using the old picasa-app (yes, and it's quick!), others ACDSee (way toooo slow and everyone has a separate db)...

Had an interesting discussion on discord about my inputs and would love to share the ideas: https://discord.com/channels/979116623879368755/1349684390577442816

![image](https://github.com/user-attachments/assets/634b4ee6-4e47-4672-9261-9b4ae2b588e9)

**Comment by bo0tzz** [comment]
Totally missed it, but this should now be supported since v1.127.0

**Reply by luziusalbin** [reply]
alright - that's the`100% dealmaker answer for me. thanks!

**Reply by tztopi** [reply]
> alright - that's the`100% dealmaker answer for me. thanks!

Have you tested this on UI? Rescan has finished on all of my external libraries, but the directory/path substring search seems not working.
Server version: v1.129.0
Image: ghcr.io/alexbelgium/immich-amd64:1.129.0
Path: /xxx/yyy/20211213_igepio/IMG_6684.JPG

The substring in search on UI: igepio
I've also tried with context search and file/extension search, but nothing relevant in the search result.

What am I doing wrong?

Edit: If I change the search URL parameter from originalFileName to originalPath
/search?query=%7B"originalFileName"%3A"igepio"%7D --->
/search?query=%7B"originalPath"%3A"igepio"%7D
works. Seems the mentioned missing UI functionality PR https://github.com/immich-app/immich/pull/16173 still an issue, theres no type selector on the UI.

**Reply by ffchung** [reply]
The PR #16173 look like only implement the backend, but still not work for fount-end.

**Reply by bo0tzz** [reply]
You're right, I've reopened this for now.

**Reply by luziusalbin** [reply]
well well - i cannot test this, because of "not having immich ready"  :D

**Reply by charlotron** [reply]
Omg! as @tztopi sad it is implemented in backend and I already tested and works, we only need the UI changes.
![image](https://github.com/user-attachments/assets/cade9b0c-98eb-47bf-b718-b4264fd063c8)

Pleaseee this is needed, it is just a damn radio and an input box!! XD

Thank you in advance!!

**Reply by luziusalbin** [reply]
This should be on top of priorities to make Immich usable for existing and already well organized photo-directories with much older dates. <3 please! :-)

**Reply by luziusalbin** [reply]
hey @bo0tzz - any updates to this?
i really would love to see the chance, that you can search for the "relative path" as an input.
thank you!

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Reply by luziusalbin** [reply]
yeah - don't want to bother... really! :-)

If i get the point right of your pasted link:
"_The most effective thing you can do is make a financial donation to the project and to the people who maintain it._" - where can i do the right donation, to receive a most quick implementation of this thread?

thanks in advance for your constructiove help, detailed informations about future updates and kind opinions, how i can help to bring that project forward as much as possible...

**Reply by luziusalbin** [reply]
> Omg! as @tztopi sad it is implemented in backend and I already tested and works, we only need the UI changes. !
> 
> Pleaseee this is needed, it is just a damn radio and an input box!! XD
> 
> Thank you in advance!!

This still works - the codes are there. no crazy wizardry needed...

what i'm missing is that it's just looking at the final path and not checking for multiple places.
Input: .../search?query=%7B"originalPath"%3A"beach"%7D  --> it show's every photo in every subfolder that contains "beach"
Input: .../search?query=%7B"originalPath"%3A"beach%20thailand"%7D  --> it show's **only** photos that exactly contains "beach thailand" in a sequence

this should be flexible enough to show photos from folder "thailand beach", "beach party thailand" or "best beach in thailand" as well.

**Comment by JGKle** [comment]
Oh man - I tried out Immich in 2022 while looking for an alternative to Synology Photos (for Android), but passed because I couldn't use my existing folder organization. So stoked to see this has been added since! The ability to search by paths *on mobile* is the one & only blocker...so close to being able to move over!!! 🙏 🙏

**Comment by bo0tzz** [comment]
WIP in #18413

**Comment by TobbeJ** [comment]
question related to this part of the original feature request:
"this could be solved by adding the "file path" of every assset/picture to the searchable items. This way I could search for a known folder name and add the result to an album of my choice / create a new album"

can the search result be turned in to an album?
i'm asking because my (external) library is a bit on the large side with several 100k photos and i cant possibly organize them by clicking one photo at a time.
but it should be quick to convert certain paths to albums

**Comment by ManiacDC** [comment]
@bo0tzz I noticed #18413 was closed, any idea why? Is there a plan to implement this another way?

**Reply by JGKle** [reply]
Same question, this is the primary feature I've been waiting for - kinda weird that it was just closed without any comment about why, & no apparent alternate PR.....?

**Reply by ManiacDC** [reply]
PR was closed because the author is no longer working on Immich and asked them not to use the code.

**Reply by luziusalbin** [reply]
This is very sad, I hope you can get this thing to work soon - I cannot find ANY of our pictures, which are well sorted by place/client and date...

**Reply by alextran1502** [reply]
@luziusalbin have you tried search by location and date range in the search filter that we have?

**Reply by luziusalbin** [reply]
Of course, but that is not exact enough. We have clients living next to each other - no chance to get this separated...

**Reply by JGKle** [reply]
I have the same issue - my collection is meticulously labeled by path + foldername (i.e. folder names like `19-03-08.0) Dubai - Day 2 (DhowWharfage, WaterBus, Souks, AlFahidi, MallOfEmirates, BurjAlArab, Madinat+Beach)`) - if I want to find photos where I visited Madinat, I just search in the file manager or Lightroom or Digikam for any photos with that term in the path. But since Immich ignores path names, I pretty much can't find anything - I just have to go to the physical system where they're stored in order to search.

**Reply by luziusalbin** [reply]
I think it's obvious and very essential to manage an existing, (over years) maintained and large (!) photo archive.

Example folder names from this week: 
2025-07-28 - Birthday Party Zurich
2025-07-29 - Hike in Zermatt 
2025-08-01 - 1. Augustparty Aarau
2025-08-03 - Hike in Montpellier
2025-08-04 - Market Arles

Now if I search for "hike 2025" I want all the photos from the 2nd and 4th folder. If I type "August 2025" it should show all photos from the third folder...

Please add this asap to your coding queue 🙏🏼

**Reply by luziusalbin** [reply]
> @luziusalbin have you tried search by location and date range in the search filter that we have?

The idea is nice, but doesn't help with old photos without GPS/exif data. Any these photos are the big part of my archive... 😊

**Reply by JGKle** [reply]
As a note (in the hopeful event that this is getting some traction), personally I only care about being able to do it from the mobile app.

**Reply by luziusalbin** [reply]
Personally i'm coming from the other side and need that on the PC/Web GUI version. :-)

**Comment by ManiacDC** [comment]
For anyone implementing this in the future, a really neat add-on for this would be able to click on a folder path in the "Folders" view and see all the files in that folder and its subfolders.

**Comment by BillClarkEarthPoint** [comment]
Yes, please add a partial folder name search for photos from external libraries!

**Comment by jkarjala** [comment]
I'd say the minimal UI support would be replacing the `originalFileName` with the `originalPath` in the REST API query which the UI makes with "File name" search option. At least with external libraries this finds files with the given word either in the path or the filename. Would a PR for such a simple fix be accepted?

**Reply by luziusalbin** [reply]
hey @jkarjala - this would be an acceptable change. actually i'm never searching for a filename, because no photo is namend other than "IMG_0001.jpg" or similiar.. :)

**Reply by JGKle** [reply]
> i'm never searching for a filename, because no photo is namend other than "IMG_0001.jpg" or similiar.. :)

That's precisely the issue here - filenames don't contain useful info, but are searchable; conversely paths contain lots of useful info, but are not searchable.

**Reply by ffchung** [reply]
As I think we can just combine search path and filename together.

**Reply by luziusalbin** [reply]
this would be 'the' solution. that's what i'm looking for...

**Reply by jkarjala** [reply]
It seems this recent PR is addressing the web UI for this feature: https://github.com/immich-app/immich/pull/26758

**Reply by luziusalbin** [reply]
oh! this is new... good to know, that there's happening something!

**Comment by arnohoff** [comment]
It would be great to be able to search words in the filepath from externa libarys in this search.
